src/web/content.rs

```diff
@@ -116,6 +122,65 @@ impl<'a, T: Serialize> Responder<'a> for Output<T> {
     }
 }
 
+/// A Rocket responder that generates response with extra HTTP headers.
+pub struct WithHttpHeaders<'h, R>(pub HeaderMap<'h>, pub Option<R>);
+
+impl<'r, R: Responder<'r>> Responder<'r> for WithHttpHeaders<'_, R> {
+    fn respond_to(self, request: &Request) -> Result<Response<'r>, Status> {
+        let mut build = Response::build();
+
+        if let Some(responder) = self.1 {
+            build.merge(responder.respond_to(request)?);
+        }
+
+        for header in self.0.iter() {
+            // Due to a lifetime markers hell, I couldn't figure out what I
+            // need to do in order to use `build.header(header)` instead.
+            // This is something I need to look into in future.
+            build.raw_header(header.name().to_owned(), header.value().to_owned());
+        }
+
+        build.ok()
+    }
+}
+
+/// A guard that implements extra validation for pagination's 'limit' value.
+/// Essentially ensures it lies within an allowed range.
+pub struct PaginationLimit(pub usize);
+
+impl FromFormValue<'_> for PaginationLimit {
```

Comment:
As noted above, let's implement `default()` for this one and remove the `Option<PaginationLimit>` from the API route (https://api.rocket.rs/v0.4/src/rocket/request/form/from_form_value.rs.html#182-191)?